core/client/fs/src/main/java/alluxio/client/file/FileSystem.java

```diff
@@ -147,7 +148,8 @@ public static FileSystem create(FileSystemContext context) {
       if (conf.getBoolean(PropertyKey.USER_CLIENT_CACHE_ENABLED)
           && CommonUtils.PROCESS_TYPE.get().equals(CommonUtils.ProcessType.CLIENT)) {
         try {
-          return new LocalCacheFileSystem(fs, conf);
+          CacheManager cacheManager = CacheManager.Factory.get(conf);
+          return new LocalCacheFileSystem(cacheManager, fs, conf);
         } catch (IOException e) {
           LOG.error("Fallback without client caching: ", e);
```

Comment:
Since `CacheManager.Factory.get` is nullable, we should put this in a null check instead of exception clause? Up to you how you want to handle the error path

Reply:
Changed `CacheManager.Factory`  so it is no more nullable